vault/data_source_aws_static_credentials.go

```diff
@@ -0,0 +1,78 @@
+package vault
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+)
+
+const staticCredsAffix = "static-creds"
+
+func awsStaticCredDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadContextWrapper(awsStaticCredentialsDataSourceRead),
+		Schema: map[string]*schema.Schema{
+			// backend is deprecated, but the other AWS resource types use it, and predate the deprecation.
+			// It's probably more helpful to the end user to maintain this consistency, in this particular case.
+			consts.FieldBackend: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "AWS Secret Backend to read credentials from.",
+			},
+			consts.FieldName: {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Name of the role.",
+			},
+			consts.FieldAccessKey: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "AWS access key ID read from Vault.",
+				Sensitive:   true,
+			},
+			consts.FieldSecretKey: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "AWS secret key read from Vault.",
+				Sensitive:   true,
+			},
+		},
+	}
+}
+
+func awsStaticCredentialsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	mount := d.Get(consts.FieldBackend).(string)
```

Comment:
nit, for consistency
```suggestion
	backend := d.Get(consts.FieldBackend).(string)
```